pkg/controller/common/helpers.go

```diff
@@ -503,6 +506,101 @@ func ParseAndConvertConfig(rawIgn []byte) (ign3types.Config, error) {
 	}
 }
 
+// Internal errors used for base64-decoding and gunzipping Ignition configs
+var (
+	errConfigNotGzipped       = fmt.Errorf("ignition config not gzipped")
+	errConfigNotBase64Encoded = fmt.Errorf("ignition config not base64 encoded")
+)
+
+// Decode, decompress, and deserialize an Ignition config file.
+func ParseAndConvertGzippedConfig(rawIgn []byte) (ign3types.Config, error) {
+	// Let's try to decode and decompress our payload
+	out, err := decodeAndDecompressPayload(bytes.NewReader(rawIgn))
+	if err == nil {
+		// Our payload was decoded and decompressed, so let's deserialize it.
+		return ParseAndConvertConfig(out)
+	}
+
+	// Our Ignition config is not base64-encoded, which means it might just be gzipped.
+	if errors.Is(err, errConfigNotBase64Encoded) {
+		out, err = decompressPayload(bytes.NewReader(rawIgn))
+		glog.V(2).Info("ignition config was not base64 encoded, trying to gunzip ignition config")
+		if err == nil {
+			return ParseAndConvertConfig(out)
+		}
+	}
+
+	// Our Ignition config is not gzipped, so let's try to serialize the raw Ignition directly.
+	if errors.Is(err, errConfigNotGzipped) {
+		glog.V(2).Info("ignition config was not gzipped, trying to parse ignition directly")
+		return ParseAndConvertConfig(rawIgn)
```

Comment:
ah so we fall back to just parsing here. I wonder if we should name the function somehow to indicate this, or split it out somehow for clarity. No strong feelings

Reply:
For now, I don't think we should split out the functionality. Until the compression / encoding code is merged into the in-place upgrade controller, we'll need to fall back to just parsing (as it works now).

In terms of renaming this function to indicate that it will fall back, I'm open to that, but lack the creativity to come up with a decent name :smile:.

Reply:
Ok, that makes sense to me! Let's just keep it as is for now, thanks!